tests/identifiers/test_datacite.py

```diff
@@ -0,0 +1,189 @@
+
+class MockDataciteClient(object):
+
+    def __init__(self, *arg, **kwargs):
+        pass
+
+    metadata_get = mock.Mock(return_value=datacite_metadata_response())
+    metadata_post = mock.Mock(return_value='OK (10.5072/FK2osf.io/yvzp4)')
+
+@pytest.fixture()
+def datacite_client():
+    return DataCiteClient()
+
+@pytest.fixture()
+def registration():
+    return RegistrationFactory()
+
+@pytest.mark.django_db
+class TestDataCiteClient:
+
+    @responses.activate
+    @mock.patch('website.identifiers.clients.datacite.DataCiteMDSClient', MockDataciteClient)
+    @mock.patch('website.settings.DATACITE_URL', 'https://mds.fake.datacite.org')
+    def test_datacite_create_identifiers(self, datacite_client, datacite_node_metadata):
+        responses.add(
+            responses.Response(
+                responses.POST,
+                settings.DATACITE_URL + '/metadata',
+                body='OK',
+                status=200
+            )
+        )
+        identifiers = datacite_client.create_identifier(datacite_node_metadata)
+
+        assert identifiers['doi'] == '10.5072/FK2osf.io/yvzp4'
+        MockDataciteClient.metadata_post.assert_called_with(datacite_node_metadata)
+
+    @responses.activate
+    @mock.patch('website.identifiers.clients.datacite.DataCiteMDSClient', MockDataciteClient)
+    @mock.patch('website.settings.DATACITE_URL', 'https://mds.fake.datacite.org')
+    def test_datacite_change_status_identifier(self, datacite_client, datacite_node_metadata):
+        responses.add(
+            responses.Response(
+                responses.POST,
+                settings.DATACITE_URL + '/metadata',
+                body='OK',
+                status=200
+            )
+        )
+
+        identifiers = datacite_client.change_status_identifier(status=None, metadata=datacite_node_metadata)
+
+        assert identifiers['doi'] == '10.5072/FK2osf.io/yvzp4'
+        MockDataciteClient.metadata_post.assert_called_with(datacite_node_metadata)
+
+    def test_datacite_build_doi(self, registration, datacite_client):
+        assert datacite_client.build_doi(registration) == settings.DOI_FORMAT.format(prefix=settings.DATACITE_PREFIX, guid=registration._id)
+
+    def test_datacite_build_metadata(self, registration, datacite_client):
+        metadata_xml = datacite_client.build_metadata(registration).encode('utf-8')
+        parser = lxml.etree.XMLParser(ns_clean=True, recover=True, encoding='utf-8')
+        root = lxml.etree.fromstring(metadata_xml, parser=parser)
+        xsi_location = '{http://www.w3.org/2001/XMLSchema-instance}schemaLocation'
+        expected_location = 'http://datacite.org/schema/kernel-4 http://schema.datacite.org/meta/kernel-4/metadata.xsd'
+        assert_equal(root.attrib[xsi_location], expected_location)
+
+        identifier = root.find('{%s}identifier' % schema40.ns[None])
+        assert_equal(identifier.attrib['identifierType'], 'DOI')
```

Comment:
For consistency with the other tests, use plain asserts rather than `assert_equal`, et al.